pom.xml

```diff
@@ -1064,6 +1065,32 @@
                 <version>${iceberg.version}</version>
             </dependency>
 
+            <!-- Hudi dependency
+            TODO: replace this 4 dependency with hudi-spark3-bundle_2.12 -->
+            <dependency>
+                <groupId>org.apache.hudi</groupId>
+                <artifactId>hudi-spark3_${scala.binary.version}</artifactId>
```

Comment:
IMO, this package is broken, the `hudi-spark3_2.12` depends on `hudi-spark-common_2.11`, which will cause the Scala library conflict, you can check at
https://mvnrepository.com/artifact/org.apache.hudi/hudi-spark3_2.12/0.9.0

Reply:
cc @yanghua